lib/src/component2_suggestors/class_name_and_annotation_migrator.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:codemod/codemod.dart';
+
+import '../constants.dart';
+import 'component2_utilities.dart';
+
+/// Suggestor that replaces `UiComponent` with `UiComponent2` in extends clauses
+/// and updates the annotation to `@Component2()`.
+///
+/// The same update is made for `UiStatefulComponent` and `react.Component`
+/// (including instances where `react.Component` is used for typing).
+/// `react.dart` imports are also updated to show Component2 as well
+/// as Component.
+class ClassNameAndAnnotationMigrator extends GeneralizingAstVisitor
+    with AstVisitingSuggestorMixin
+    implements Suggestor {
+  final bool noPartialUpgrades;
```

Comment:
#nit consider changing `noPartialUpgrades` to `allowPartialUpgrades` or something similar.

There are conventions to help boolean names be more readable / logical. Dart has some best practices [here](https://dart.dev/guides/language/effective-dart/design#consider-omitting-the-verb-for-a-named-boolean-parameter) and we have a list of best prefixes [here](https://github.com/Workiva/over_react_analyzer_plugin/blob/link-2019-integration/over_react_analyzer_plugin/lib/src/diagnostic/bool_prop_name_readability.dart#L96).

Reply:
Thank you for the resources. I'll change that!